src/index.tsx

```diff
@@ -36,23 +37,22 @@ type setStored<Kind> = (newValue: Kind) => void
  * @param name The key used to store the data in.
  * @param InputType An optional type used for the stored value. This is only a type assertion on the input. The output type will always be a value returned by Immutable.fromJS.
  */
-export const useStorage = <InputType extends {}>(name: string, {
-   /**
-    * The default value if no stored value is found.
-    */
-    placeholder,
-
-   /**
-    * The Storage used. Defaults to localStorage.
-    */
-    storageArea = window.localStorage
-} : {
-    placeholder: InputType, storageArea: Storage
-  }): [immutableValue, setStored<InputType>] => {
+export const useStorage = <InputType extends {}>(
+    name: string,
+    { placeholder, storageArea = window.localStorage } : Options<InputType> = {}
```

Comment:
Adding the trailing `= {}` default value here is the key fix for the first issue